<!-- -*- mode: XML -*- -->

<launch>
   <!-- This launch roscore -->
   <master auto="start"/>

  <!--static TFs -->
  <include file="$(find teresa_driver)/launch/static_tfs.launch" />

  <!-- IMU -->
  <node pkg="receive_xsens" type="receive_xsens" name="receive_xsens" output="screen" respawn="true" />


  <!-- teresa driver -->  

   <node pkg="teresa_driver" type="teresa_node" name="teresa_node" output="screen" respawn="false">
   <param name="board1" type="str" value="/dev/serial/by-id/usb-FTDI_MM232R_USB_MODULE_board1-if00-port0"/>
   <param name="board2" type="str" value="/dev/serial/by-id/usb-FTDI_MM232R_USB_MODULE_FT94LE8L-if00-port0"/>
   <param name="base_frame_id" type="str" value="/base_link"/>	  
   <param name="odom_frame_id" type="str" value="/odom"/>  
   <param name="head_frame_id" type="str" value="/teresa_head"/>
   <param name="stalk_frame_id" type="str" value="/stalk_head"/>                  
   <param name="freq" type="double" value="20.0" />               
   <param name="using_imu" type="int" value="1"/>                   
   <param name="simulation" type="int" value="0"/>
   <param name="publish_temperature" type="int" value="1"/>
   <param name="publish_buttons" type="int" value="1"/>
   <param name="publish_volume" type="int" value="1"/>
   <param name="publish_diagnostics" type="int" value="1"/>
   <param name="number_of_leds" type="int" value="38"/>
   <param name="inverse_left_motor" type="bool" value="true"/>
   <param name="inverse_right_motor" type="bool" value="false"/>	
   <param name="initial_dcdc_mask" type="int" value="255"/> <!--243 lasers off-->  
   <param name="final_dcdc_mask" type="int" value="255"/>  <!--243 lasers off -->
   <param name="height_velocity" type="int" value="20"/>
   <param name="tilt_velocity" type="int" value="2"/>
   <param name="leds_pattern" type="str" value="knight_rider"/>	
   </node>

   <!-- joy node -->
   <node respawn="true" pkg="joy" type="joy_node" name="joy_node">
   <!--<param name="dev" value="/dev/input/js1"/>-->
   </node>
  
    <!-- teresa joystick driver -->
   <node pkg="teresa_driver" type="teresa_teleop_joy" name="teleop" output="screen" respawn="false">
	<remap from="cmd_vel" to="teresa/cmd_vel"/>
   </node>


   <node pkg="assisted_steering" name="assisted_steering" type="assisted_steering" output="screen">
	<param name="laser_topic" type="string" value="scan360" />
	<param name="odom_topic" type="string" value="odom" />
	<param name="cmdvel_topic" type="string" value="teresa/cmd_vel" />
	<param name="new_cmdvel_topic" type="string" value="cmd_vel" />
	<param name="max_lin_vel" type="double" value="0.5" />
	<param name="max_ang_vel" type="double" value="0.8" />
	<param name="max_lin_acc" type="double" value="1.0" />
	<param name="max_ang_acc" type="double" value="1.0" />
	<param name="time_step" type="double" value="0.5" />
	<param name="granularity" type="double" value="0.025" />
	<param name="robot_radius" type="double" value="0.32" />
	<param name="is_active" type="bool" value="true" />
	
  </node>


   <!-- Xtions -->
   <!--<include file="$(find teresa_driver)/launch/xtions.launch" />-->

</launch>
